### DISCUSSION TITLE
Placement: how to define placement in my Custom Theme, and how to know existent placement in other Themes

### DISCUSSION THREAD

**Discussion by MarGraz** [OrchardCMS/OrchardCore#15464] [Q&A]
Hello everyone,

I'm diving into the world of "Placement" but I need some clarification. This topic is addressed in the [documentation](https://docs.orchardcore.net/en/latest/docs/reference/core/Placement/#placement-files), as well as in the "Training Demo" module, specifically in the "[Display Management](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/Lombiq.TrainingDemo/Controllers/DisplayManagementController.cs)" section. Additionally, I've watched [this video](https://youtu.be/yDtUyQpormU) from the Dojo Course 3 for further insights.

To give you a brief overview, I've implemented my `MyTextFieldDisplayDriver : ContentFieldDisplayDriver<MyTextField>` within my `CustomTheme` module. I used the **`ocfielddriverfull`** [code snippet](https://www.youtube.com/watch?v=5L95C0qxrA0), I've managed to utilize the `Location()` method ([as demonstrated here](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Menu/Drivers/ContentMenuItemPartDisplayDriver.cs#L20)) in the `Display` and `Edit` override methods, to specify the placement of my field shape. However, the process for defining these positions remains unclear to me 😶‍🌫️ 

This is why I have several questions:

1. I understand that Modules can have a `Placement.json`, but I'm wondering how `Placement` can be defined within a custom theme. Specifically, how can I tag specific parts for use, such as `"Admin", "Content:10"` (as showed [here](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Menu/Drivers/ContentMenuItemPartDisplayDriver.cs#L20)), I need first to define it in my `Layout.cshtml`?
2. Is there a way to know the available placements of modules developed by others?
3. How can I determine the placement options available in the Admin backoffice UI, especially since `MyTextFieldDisplayDriver` also uses a `Location()` method in the `Edit` override?
4. Do themes typically come with graphical maps of all available placements, similar to the "[Module Map](https://docs.joomla.org/Module_Position#Module_Map)" used by Joomla? This seems like a potentially useful feature that could be added to each theme in the theme gallery to show the available positions in backend and frontend themes.

Here an example of Joomla "Module Map" feature:

![image](https://github.com/OrchardCMS/OrchardCore/assets/19241483/fe93e1b0-da52-43e0-9e87-4a0a4f4201cc)

Thank you so much for any reply

**Comment by MarGraz** [comment]
Maybe what I'm looking for is "**display types**", as explained in the [documentation about zones differentiators](https://docs.orchardcore.net/en/latest/docs/reference/modules/Themes/#content-zones-differentiators). There's a `Content-{Content Item name type}.cshtml` file that describes the page. However, I'm wondering **why there is no reference or link redirecting to the "display types" page on the "placement"** [page](https://docs.orchardcore.net/en/main/docs/reference/core/Placement/). Are they two different concepts? 🤔 

For what I've learned, it seems that the positioning described by a "Theme", takes precedence over templates and other definitions, as [described here](https://docs.orchardcore.net/en/latest/docs/reference/modules/Themes/#content-zones-differentiators:~:text=Templates%20are%20usually%20provided%20by%20the%20modules%20that%20create%20the%20corresponding%20shape%20types,%20but%20can%20always%20be%20redefined%20by%20a%20Theme.).

**Questions are more or less the same:**

1. To achieve my goal of styling shapes in a specific way and positioning them inside my custom theme, what is the best practice? Is better to use "display types" with `Content-{...}.cshtml`, or use `Placement.json`?
2. In my `DisplayDriver`, how can I define specific parts to use, such as `"Admin", "Content:10"` (as shown [here](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Menu/Drivers/ContentMenuItemPartDisplayDriver.cs#L20)) in the `Location()` method?
3. Regarding the Admin UI, how can I position my custom field in a specific place if I don't know the available positions in the Admin theme?

Thank you

**Reply by MarGraz** [reply]
Hi everyone, could someone please assist me with the questions I asked in this post? 😊

Thank you

**Comment by MarGraz** [comment]
Hi @aaronamm, as mentioned in the chat during the YouTube theme development live [session](https://www.youtube.com/watch?v=RdE6Io4XiZ8), here is my discussion about the `Placement`. 

Thank you 😊

**Comment by hishamco** [comment]
Why not use [zones](https://docs.orchardcore.net/en/latest/reference/modules/Layers/#zones)?

**Reply by MarGraz** [reply]
@hishamco thank you for your reply 😊

I never tried [zones](https://docs.orchardcore.net/en/latest/reference/modules/Layers/#zones). Do they also work with Razor views, or they are specific for `widgets`?

What I'm looking for is to **define precise areas** inside my `Layout.cshtml` (or similar place) and assign it `names`, where I can display `Shape` via `MyCustomDisplayDriver`. Right now, it's not clear to me how to define names for these areas.

Thank you

**Reply by hishamco** [reply]
You can render zones in Razor using `@await RenderSectionAsync("ZoneName", required: false)`

For example:

https://github.com/OrchardCMS/OrchardCore/blob/768459c5a25ac05aa2f0304353892b4ff434444c/src/OrchardCore.Themes/TheAdmin/Views/Layout.cshtml#L81C39-L81C52

Usage:

https://github.com/OrchardCMS/OrchardCore/blob/768459c5a25ac05aa2f0304353892b4ff434444c/src/OrchardCore.Modules/OrchardCore.Workflows/Views/Workflow/Details.cshtml#L8-L17
